### ISSUE TITLE
Rider Support

### ISSUE THREAD

**Issue by crunchy234** [MrJul/ForTea#89]
Now that Rider has been released I only use Visual Studio for the things I can't do in Rider.

I was wondering if you were planning to release ForTea for Rider since it is also a Jetbrains product?

**Comment by kendallb** [comment]
Same here!

**Comment by khionu** [comment]
Bumping for support.

**Comment by matkoch** [comment]
@MrJul there is VisualStudio SDK code, right? So I assume that Rider support isn't possible atm.

**Comment by eltiare** [comment]
Oh sad. I was really hoping for some sort of way to use this plugin. :(

**Comment by kendallb** [comment]
I love this extension and would love it for Rider. One thing I hate about this for R# is the stupid versioning system R# uses, in that once they bump to a new minor revision the extensions have to be updated! Grrr. And ForTea still has not been re-released since R# 2018.2 has come out. 

JetBrains really needs to fix this, as ForTea will work fine with the latest R# releases as soon as the version is bumped. 

Maybe the ForTea folks should add meta data in there saying it works with the next 10 future versions. Ie: 2018.2, 2018.3, 2018.4 ... 2018.10!

**Comment by genyx** [comment]
Since Rider grows so fast and is very powerful now, i would like to see this plugin in Rider too. 👍

**Comment by nvirth** [comment]
+1

**Comment by kendallb** [comment]
Any idea if this can be ported to Rider?

**Comment by zgabi** [comment]
Do you know what is this
https://github.com/MrJul/ForTea/pull/102
and this?
https://github.com/MrJul/ForTea/pull/103

It seems that the guy was working a lot on it, then deleted.

**Comment by kirillgla** [comment]
Yes, I am working on T4 support in Rider. Sorry, I opened those pull requests by accident; they were supposed to be pull requests into my repository.
My fork is not open-source atm but will become open-source soon

**Comment by kirillgla** [comment]
Btw the first version of the plugin for `Rider 2019.3 EAP1` [is now available](https://github.com/MrJul/ForTea/files/3804735/ForTea.Frontend-2019.3.zip)!
Pretty much everything from the original plugin now works in Rider.
To install this plugin, go `Settings > Plugins > Install plugin from disk`.
I encourage you to try it. The plugin is still in its early days, so any feedback is welcome. Thank you!

**Comment by kirillgla** [comment]
P.S.: Please note that the plugin does not support EnvDTE library yet.
P.S.S: Because Rider already has T4 executor by default (but a one that cannot expand macros), you most likely want to disable that. To do that, go `Settings > Tools > Custom Tools > T4 Template > Disable` (red `x` to the right)

**Comment by kendallb** [comment]
That's awesome, I will give it a try. Does it require EAP1? I hope the Rider version won't be locked to the versioning of Rider itself like the R# one is, as that is super annoying that every time R# gets updated you need a new build on ForTea, and literally all that is done is to add the new version and recompile?

**Comment by kendallb** [comment]
BTW, I did not think EnvDTE worked at all in Rider anyway? I have been re-writing my templates to get rid of it.

**Comment by kirillgla** [comment]
I am afraid it is not possible for a plugin to target multiple versions of Rider right now, so yes, that plugin would require `EAP 1`. [Here is](https://github.com/MrJul/ForTea/files/3807886/ForTea.Frontend-2019.2.3.zip) a plugin for `Rider 2019.2.3` in case you don't want to install the EAP.
And updating plugin is, unfortunately, more than just adjusting the version because Rider (as well as R#) does not guarantee backwards compatibility for plugins

**Comment by kirillgla** [comment]
You are right about `EnvDTE`, it doesn't work at all in Rider. There are some vague plans on making it work in T4 in Rider, but that is unlikely to happen anytime soon

**Comment by kendallb** [comment]
Well thats annoying about the plugins. Maybe there needs to be some binary compatibility for the plugins at some point that is maintained, at least across major versions? It's a bit silly to be requiring new versions for every minor IDE update?

Although if the plugin is actively maintained and updated it is not a problem, but that was not happening for the R# one.

**Comment by kendallb** [comment]
The one for 2019.2.3 works great so far!

**Comment by kirillgla** [comment]
The plugin is now installed in Rider out of the box, so I believe this issue is solved and can be closed?

**Comment by kendallb** [comment]
Yeah probably. I have not tried the EAP. Waiting on the release version to try it. Do you know when that will be arriving?

**Comment by matkoch** [comment]
@kendallb I advice you to try the EAP. If everyone was about to only try the release version, there is no point in having an EAP. Especially when new features arrive.

**Comment by kendallb** [comment]
Yeah I know I have used EAP's in the past, but we are in code crunch time at the moment so I don't have time to deal with issues. Although EAP3 is probably pretty solid. Maybe later in the week.
